### DISCUSSION TITLE
Prefix the zod validators with a Z

### DISCUSSION THREAD

**Discussion by roninjin10** [wagmi-dev/abitype#219] [Ideas]
Because zod validators have the same name as the typescript type it is inconvenient to use both in the same file. Solution is to prefix them with a z like ZAddress. This would be a breaking change though for whenever there is going to be a major version bump

**Comment by Raiden1411** [comment]
I would prefer something more explicit like ZodAddress but I agree that it would make things simpler than having to do something like `import {Abi as ZodAbi} from "abitype/zod"`

**Reply by roninjin10** [reply]
I'd be on board for this

**Comment by tmm** [comment]
Is this a common pattern across the Zod ecosystem? I don't use Zod often and it's not the core focus of this library. We could do this in a non-breaking way by exporting new names alongside deprecating the existing exports in a patch version (haven't hit 1.0 yet so patch === minor).

**Reply by roninjin10** [reply]
I don't think it's super common

**Comment by roninjin10** [comment]
I thought about this more and I think you should get rid of the zod validators tbh next breaking release. If you wanted to provide this functionality instead using a well typed vanilla js function like `isAddress` and `isAbi` that cast the type works better and then can compose with zod nicely

const zAddress = z.object().refine(isBytes) <- Zod is smart enough to still infer the bytes type here
